tests/identity.rs

```diff
@@ -267,3 +267,165 @@ fn refresh() {
 
     assert_eventually!(refreshed.load(Ordering::SeqCst) == true);
 }
+
+#[test]
+fn orig_dst_client_can_connect_to_tls_server_with_force_id_header() {
+    let _ = trace_init();
+
+    let proxy_name = "foo.ns1.serviceaccount.identity.linkerd.cluster.local";
+    let proxy_identity = identity::Identity::new("foo-ns1", proxy_name.to_string());
+
+    let app_name = "bar.ns1.serviceaccount.identity.linkerd.cluster.local";
+    let app_identity = identity::Identity::new("bar-ns1", app_name.to_string());
+
+    // Make a server that expects `app_identity` identity
+    let srv = server::http2_tls(app_identity.server_config)
+        .route("/", "hello")
+        .run();
+
+    // Make a proxy that has `proxy_identity` identity
+    let proxy = proxy::new()
+        .outbound(srv)
+        .identity(proxy_identity.service().run())
+        .run_with_test_env(proxy_identity.env);
+
+    // Make a non-TLS client
+    let client = client::http2(proxy.outbound, app_name);
+
+    // Assert a request to `srv` with incorrect `l5d-force-id` header fails
+    //
+    // Fails because of reconnect backoff; results in status code 502
+    assert_eq!(
+        client
+            .request(
+                client
+                    .request_builder("/")
+                    .header("l5d-force-id", "hey-its-me")
+                    .method("GET")
+            )
+            .status(),
+        http::StatusCode::SERVICE_UNAVAILABLE
```

Comment:
When a request is routed to `SO_ORIGINAL_DST`, the request seems to be bypassing the `force_identity_check::layer()` and therefore trying to connect to an identity that is incorrect or does not exist. The connection goes into a backoff and returns a `SERVICE_UNAVAILABLE`. I don't think this is the behavior we want, but maybe it is? It's an open question that I'm not totally sure how to resolve right now.